processor/filterprocessor/filter_processor_logs.go

```diff
@@ -70,31 +96,97 @@ func createLogsMatcher(lp *LogMatchProperties) (filtermatcher.AttributesMatcher,
 		filterset.Config{
 			MatchType: filterset.MatchType(lp.LogMatchType),
 		},
-		lp.ResourceAttributes,
+		getFilterConfigForMatchLevel(lp, matchLevel),
 	)
 	if err != nil {
 		return attributeMatcher, err
 	}
 	return attributeMatcher, nil
 }
 
+func getFilterConfigForMatchLevel(lp *LogMatchProperties, m MatchLevelType) []filterconfig.Attribute {
+	switch m {
+	case ResourceLevelMatch:
+		return lp.ResourceAttributes
+	case RecordLevelMatch:
+		return lp.RecordAttributes
+	default:
+		return nil
+	}
+}
+
 func (flp *filterLogProcessor) ProcessLogs(ctx context.Context, logs pdata.Logs) (pdata.Logs, error) {
-	logs.ResourceLogs().RemoveIf(func(rm pdata.ResourceLogs) bool {
+	rLogs := logs.ResourceLogs()
+
+	// Filter logs by resource level attributes
+	rLogs.RemoveIf(func(rm pdata.ResourceLogs) bool {
 		return flp.shouldSkipLogsForResource(rm.Resource())
 	})
 
-	if logs.ResourceLogs().Len() == 0 {
+	// Filter logs by record level attributes
+	flp.filterByRecordAttributes(rLogs)
+
+	if rLogs.Len() == 0 {
 		return logs, processorhelper.ErrSkipProcessingData
 	}
 
 	return logs, nil
 }
 
+func (flp *filterLogProcessor) filterByRecordAttributes(rLogs pdata.ResourceLogsSlice) {
+	for i := 0; i < rLogs.Len(); i++ {
+		ills := rLogs.At(i).InstrumentationLibraryLogs()
+
+		for j := 0; j < ills.Len(); j++ {
+			ls := ills.At(j).Logs()
+
+			ls.RemoveIf(func(lr pdata.LogRecord) bool {
+				return flp.shouldSkipLogsForRecord(lr)
+			})
+		}
+
+		ills.RemoveIf(func(ill pdata.InstrumentationLibraryLogs) bool {
+			return ill.Logs().Len() == 0
+		})
+	}
+
+	rLogs.RemoveIf(func(rl pdata.ResourceLogs) bool {
+		return rl.InstrumentationLibraryLogs().Len() == 0
+	})
+}
+
+// shouldSkipLogsForRecord determines if a log record should be processed.
+// True is returned when a log record should be skipped.
+// False is returned when a log record should not be skipped.
+// The logic determining if a log record should be skipped is set in the
+// record attribute configuration.
+func (flp *filterLogProcessor) shouldSkipLogsForRecord(lr pdata.LogRecord) bool {
+	if flp.includeRecords != nil {
+		matches := flp.includeRecords.Match(lr.Attributes())
+		if !matches {
+			return true
+		}
+	}
+
+	if flp.excludeRecords != nil {
+		matches := flp.excludeRecords.Match(lr.Attributes())
+		if matches {
+			return true
+		}
+	}
+
+	return false
+}
+
 // shouldSkipLogsForResource determines if a log should be processed.
 // True is returned when a log should be skipped.
 // False is returned when a log should not be skipped.
 // The logic determining if a log should be skipped is set in the resource attribute configuration.
 func (flp *filterLogProcessor) shouldSkipLogsForResource(resource pdata.Resource) bool {
+	if flp.includeResources == nil && flp.excludeResources == nil {
+		return false
+	}
+
```

Comment:
This part looks unrelated to the PR's goal. 

And is there any benefit from this change?

Reply:
Indeed, it should have ended up here. Thanks! Removed.